experimental/dds/tree2/src/util/typeCheck.ts

```diff
@@ -148,13 +148,12 @@ export type requireFalse<_X extends false> = true;
 /**
  * Returns a type parameter that is true iff Source is assignable to Destination.
  *
+ * @privateRemarks
+ * Use of [] in the extends clause prevents unions from being distributed over this conditional and returning `boolean` in some cases.
+ * @see [distributive-conditional-types](https://www.typescriptlang.org/docs/handbook/2/conditional-types.html#distributive-conditional-types) for details.
```

Comment:
To better adhere to TSDoc spec
```suggestion
 * @see {@link https://www.typescriptlang.org/docs/handbook/2/conditional-types.html#distributive-conditional-types | distributive-conditional-types} for details.
```

Reply:
Markdown links are supported by some tools, but to my knowledge are not supported by TSDoc.